rfcs/0118-optional-final-semicolon.md

```diff
@@ -0,0 +1,80 @@
+---
+feature: optional-final-semicolon
+start-date: 2021-12-03
+author: Lucas Eduardo Wendt
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Make it acceptable a semicolon at the end of a Nix expression raising a warning instead of a syntax error.
+
+# Motivation
+[motivation]: #motivation
+
+Faster iterations.
+
+Some people use full keyboard editors and hack around with their configurations, sometimes you move code from the final
+part to a let expression or the reverse to test something or debug some faulty part then forgot a semicolon on the 
+final of the last line, the code fails with a syntax error then you reopen the editor just to remove the semicolon to
+make the parser happy, then you forgot the semicolon when you put the thing back to a let expression. This should be
+a recoverable syntax issue just raising a warning to remove that semicolon when all is over.
+
+This would make iterations a little faster and bring a bit less friction, and give a better clue about what is going on in this case.
+
+# Detailed design
+[design]: #detailed-design
+
+Allow an optional semicolon at the end of an expression raising a warning if it's provided.
+
```

Comment:
This is not a detailed design.

I would expect to get a good idea of how this will be implemented technically.

Note that I tried adding an optional semicolon to the parser a few years ago, but it was rejected because my implementation was naïve (just have a giant lookahead essentially). So I’d like to see a technical design here that is feasiblae.